src/rpc/mining.cpp

```diff
@@ -386,9 +386,9 @@ static RPCHelpMan generateblock()
         block.vtx.insert(block.vtx.end(), txs.begin(), txs.end());
         RegenerateCommitments(block, chainman);
 
-        BlockValidationState state;
-        if (!TestBlockValidity(state, chainman.GetParams(), chainman.ActiveChainstate(), block, chainman.m_blockman.LookupBlockIndex(block.hashPrevBlock), /*fCheckPOW=*/false, /*fCheckMerkleRoot=*/false)) {
-            throw JSONRPCError(RPC_VERIFY_ERROR, strprintf("TestBlockValidity failed: %s", state.ToString()));
+        std::string reason;
+        if (!chainman.TestBlockValidity(block, reason, /*check_pow=*/false, /*check_merkle_root=*/false)) {
+            throw JSONRPCError(RPC_VERIFY_ERROR, strprintf("TestBlockValidity failed: %s", reason));
```

Comment:
In commit "validation: refactor TestBlockValidity" (1d029c23a143f1b9110377967a7c4a20ee75058d)

There seems to be a change in behavior here if `ValidationState::m_debug_message` is set because now it would previous be included in `ToString` output but now will be dropped.

Reply:
I replaced the four `state.GetRejectReason()` with `state.ToString()`, as it was in the original.

Reply:
I changed `TestBlockValidity` to also take a `debug` argument.